src/main/java/com/example/controller/SimulationController.java

```diff
@@ -315,6 +328,7 @@ public void doStepTask() {
             processStep();
 
             if (isFinished.get()) {
+                loggerController.addItem("[Finished] Simulation finished");
```

Comment:
I'd add `loggerController.addItem("")` before `[Finished]` logs. Also as Marcin mentioned before we could move code that is inside `if` statement to another method callled "onFinished" since we reapeat it.